File: ProjectEuler/Cpp/002.cpp
#include <iostream>

using namespace std;

int main()
{
    int a = 1;
    int b = 2;
    int total = 0;

    while (b <= 4000000)
    {
        if (b % 2 == 0)
        {
            total += b;
        }
        b += a;
        a = b - a;
    }
    cout << total << endl;
}

File: ProjectEuler/Rust/src/bin/002.rs
fn main() {
    let mut first = 1i32;
    let mut second = 2i32;
    let mut total = 2i32;

    while second <= 4000000 {
        let temp = first + second;
        if &temp % 2 == 0 {
            total += temp;
        }
        second = temp;
        first = temp - first;
    }

    println!("{}", total) // Outputs 4613732
}
